Bump version to 0.0.18 (#2869)
astral-sh/ty: CHANGELOG.md

# Changelog

## 0.0.18

Released on 2026-02-20.

### Bug fixes

> - Support classes dynamically created via `type(...)` with cyclic bases ([#22792](https://github.com/astral-sh/ruff/pull/22792))

Author: There are examples in the mdtests in that PR that work fine at runtime, e.g. `X = type("X", (NamedTuple("NT", [("field", "X | int")]),), {})`

- Fix incorrect types inferred when unpacking mixed tuples ([#23437](https://github.com/astral-sh/ruff/pull/23437))
- Fix stack overflow for self-referential `TypeOf` in annotations ([#23407](https://github.com/astral-sh/ruff/pull/23407))
- Visit parameters in source order ([#23403](https://github.com/astral-sh/ruff/pull/23403))
- Visit pattern arguments in source order ([#23398](https://github.com/astral-sh/ruff/pull/23398))

> - Visit rest parameters in patterns in source order ([#23401](https://github.com/astral-sh/ruff/pull/23401))

Reviewer: ```suggestion
- Fix several server panics that could occur when computing semantic tokens for the current file ([#23403](https://github.com/astral-sh/ruff/pull/23403)), [#23398](https://github.com/astral-sh/ruff/pull/23398), [#23401](https://github.com/astral-sh/ruff/pull/23401))
```

- Exclude test-related symbols from non-first-party packages in auto-import completions ([#23252](https://github.com/astral-sh/ruff/pull/23252))
- Fix bug where diagnostics could disappear after opening an external file ([#23447](https://github.com/astral-sh/ruff/pull/23447))
- Omit loop header definitions in Go-To Definition ([#23391](https://github.com/astral-sh/ruff/pull/23391))
- Qualify inlay hint edit symbol when possibly referencing another variable ([#23265](https://github.com/astral-sh/ruff/pull/23265))
- Resolve TypeVars in `call_signature_details` parameter types ([#23149](https://github.com/astral-sh/ruff/pull/23149))

### CLI

- Add `--output-format` to `ty version` ([#23387](https://github.com/astral-sh/ruff/pull/23387))

### Configuration

- Add `replace-imports-with-any` option ([#23122](https://github.com/astral-sh/ruff/pull/23122))
- Support shellexpand for configuration paths ([#23274](https://github.com/astral-sh/ruff/pull/23274))

### Type checking

- Add a new diagnostic to detect invalid class patterns in `match` statements ([#22939](https://github.com/astral-sh/ruff/pull/22939))

> - Allow `Self` in `ClassVar` type annotation ([#23362](https://github.com/astral-sh/ruff/pull/23362))

Reviewer: ```suggestion
- Allow `Self` in `ClassVar` type annotations ([#23362](https://github.com/astral-sh/ruff/pull/23362))
```

- Consider synthesized methods and `ClassVar`-qualified declarations when determining whether an abstract method has been overridden in a subclass ([#23381](https://github.com/astral-sh/ruff/pull/23381))
- Add a diagnostic when combining `Final` and `ClassVar` ([#23365](https://github.com/astral-sh/ruff/pull/23365))
- Fix `invalid-match-pattern` false positives ([#23338](https://github.com/astral-sh/ruff/pull/23338))
- Fix return type of `assert_never` ([#23389](https://github.com/astral-sh/ruff/pull/23389))
- Fix `assert_type` diagnostic messages ([#23342](https://github.com/astral-sh/ruff/pull/23342))

> - Flag illegal special form types in PEP 613 type aliases ([#23444](https://github.com/astral-sh/ruff/pull/23444))

Reviewer: ```suggestion
- Ban PEP-613 type alias values from containing type-qualifier special forms ([#23444](https://github.com/astral-sh/ruff/pull/23444))
```

- Infer `LiteralString` for `f"{literal_str_a} {literal_str_b}"` ([#23346](https://github.com/astral-sh/ruff/pull/23346))
- Infer precise types for bit-shift operations on integer literals ([#23301](https://github.com/astral-sh/ruff/pull/23301))
- Make `[abstract-method-in-final-class]` diagnostics less verbose for classes with many abstract methods ([#23379](https://github.com/astral-sh/ruff/pull/23379))

> - Improve diagnostics for abstract final classes ([#23376](https://github.com/astral-sh/ruff/pull/23376))

Reviewer: ```suggestion
- Improve diagnostics for abstract `@final` classes ([#23376](https://github.com/astral-sh/ruff/pull/23376))
```

- Only perform literal promotion for implicitly inferred literals ([#23107](https://github.com/astral-sh/ruff/pull/23107))
- Parenthesize callable types when they appear in the return annotation of other callable types ([#23327](https://github.com/astral-sh/ruff/pull/23327))
- Consider a call to a generic function returning `Never` to terminate control flow ([#23419](https://github.com/astral-sh/ruff/pull/23419))
- Support calls to intersection types ([#22469](https://github.com/astral-sh/ruff/pull/22469))
- Validate annotated assignments to attributes on self ([#23388](https://github.com/astral-sh/ruff/pull/23388))
- Treat bytes literal as a sequence of integers ([#23329](https://github.com/astral-sh/ruff/pull/23329))
- Fix inferring type variable from string literal argument ([#23326](https://github.com/astral-sh/ruff/pull/23326))
- Support narrowing from Callable returning type guard ([#23280](https://github.com/astral-sh/ruff/pull/23280))

### Performance

> - Fall back to ambiguous for very large control flow graphs ([#23399](https://github.com/astral-sh/ruff/pull/23399))

Reviewer: ```suggestion
- Consider all code paths as being ambiguously reachable in cases with pathologically large control-flow graphs ([#23399](https://github.com/astral-sh/ruff/pull/23399))
```

### Typeshed

> - Sync vendored typeshed stubs ([#23279](https://github.com/astral-sh/ruff/pull/23279))

Reviewer: anyway, the link you're looking for is https://github.com/python/typeshed/compare/fa659b1def704dea3dc8e25c7857b23eac69df4d...1b3cec156330a93f6bb22b6636bca38c27f8f721

> - Sync vendored typeshed stubs ([#23279](https://github.com/astral-sh/ruff/pull/23279))

Author: Ok, added - thanks for the link
